store/src/java/com/zimbra/cs/extension/ExtensionDispatcherServlet.java

```diff
@@ -129,19 +136,17 @@ public void service(HttpServletRequest req, HttpServletResponse resp) throws IOE
      */
     public static void unregister(ZimbraExtension ext) {
         synchronized(sHandlers) {
-            for (Iterator it = sHandlers.keySet().iterator(); it.hasNext(); ) {
-                String path = (String) it.next();
-                if (path.startsWith("/" + ext.getName())) {
-                    ExtensionHttpHandler handler = (ExtensionHttpHandler) sHandlers.get(path);
-                    try {
-                        handler.destroy();
-                    } catch (Exception e) {
-                        ZimbraLog.extensions.warn("Error in destroy for handler " + handler.getClass(), e);
-                    } finally {
-                        it.remove();
-                    }
+            String extName = ext.getName();
+            for (ExtensionHttpHandler extHandler: EXTENSION_HANDLERS.get(extName)) {
+                try {
+                    extHandler.destroy();
+                } catch (Exception e) {
+                    ZimbraLog.extensions.warn("Error in destroy for handler " + extHandler.getClass(), e);
```

Comment:
`+` operator for string concatenation is an inefficient way to concatenate, it creates new string objects each time, leading to a lot of object creation and garbage collection, which we have seen a lot of problems of memory leaks in Zimbra in the past. And `+` operator can be slower than other methods due to the object creation and garbage collection overhead.

Reply:
Good catch